plugins/inputs/logfile/fdlimit/fdlimit_test.go

```diff
@@ -0,0 +1,21 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package fdlimit
+
+import (
+	"github.com/stretchr/testify/assert"
+	"runtime"
+	"testing"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/config"
+)
+
+func TestFileDescriptorsLimit(t *testing.T) {
+	currentOpenFileLimit, err := CurrentOpenFileLimit()
+	assert.NoError(t, err)
+
+	if runtime.GOOS == config.OS_TYPE_WINDOWS {
+		assert.Equal(t, 16384, currentOpenFileLimit)
```

Comment:
What happens if this changes on a newer version of Windows?

Reply:
`16384` has been a hard limit from 2006 (based on MySQL issues, maybe before Windows 7) and has been increased from `2048`. Therefore, I expect Windows do not decrease the file descriptors and if it is increased, it does not bring any impacts to CloudWatchAgent and as a customer, `16384` is not feasible to reach and if it is reachable , I will worry about their memory more than the CloudWatchAgent